### ISSUE TITLE
The private key cannot be used to decrypt a message

### ISSUE THREAD

**Issue by jaypi95** [passbolt/passbolt_api#191]
I'm trying to install Passbolt and I ran into a few issues so far. 
The first issue was that the Fingerprint apparently didn't match the fingerprint in the app.php.
However, after generating a new key this works.

The issue I'm getting now is that the private key can't be used to decrypt a message. Also when I try to access the site I only see a cake php file (front-controller?).

Here is my healthcheck:

---------------------------------------------------------------
 Healthcheck shell
---------------------------------------------------------------

 Environment

 [PASS] PHP version 7.0.22-0ubuntu0.16.04.1
 [PASS] PCRE compiled with unicode support
 [PASS] The temporary directory and its content are writable
 [PASS] The public image directory and its content are writable

 Config files

 [PASS] The core config file is present
 [PASS] The database config file is present
 [PASS] The email config file is present
 [PASS] The application config file is present

 Core config

 [PASS] Debug mode is off.
 [PASS] Cache is working.
 [PASS] Unique value set for security.salt
 [PASS] Unique value set for security.cipherSeed
 [PASS] Full base url is set to https://my.domain.com
 [PASS] App.fullBaseUrl validation OK.
 [FAIL] Could not reach the /healthcheck/status with the url specified in App.fullBaseUrl
  [HELP] Check that the domain name is correct in app/Config/core.php
  [HELP] Check the network settings

 SSL Certificate

 [FAIL] SSL peer certificate does not validate
 [FAIL] Hostname does not match when validating certificates.
 [WARN] Using a self-signed certificate

 Database

 [PASS] Configured to use a supported database backend
 [PASS] The application is able to connect to the database
 [PASS] Not using a prefix for database tables
 [FAIL] No table found
  [HELP] Run the install script to install the database tables
  [HELP] sudo su -s /bin/bash -c "/var/www/passbolt/app/Console/cake install" www-data
 [FAIL] No default content found
  [HELP] Run the install script to set the default content such as roles and permission types
  [HELP] sudo su -s /bin/bash -c "/var/www/passbolt/app/Console/cake install" www-data
 [PASS] The database schema up to date.

 GPG Configuration

 [PASS] PHP GPG Module is installed and loaded
 [PASS] The server gpg key is not the default one
 [PASS] The environment variable GNUPGHOME is set to /home/www-data/.gnupg
 [PASS] The directory /home/www-data/.gnupg containing the keyring is writable by the user the webserver is running as.
 [PASS] The public key file is defined in app/config.php and readable.
 [PASS] The private key file is defined in app/config.php and readable.
 [PASS] The server key fingerprint matches the one defined in app/config.php.
 [PASS] The server key defined in the app/Config.php is in the keyring.
 [PASS] There is a valid email id defined for the server key.
 [PASS] The public key can be used to encrypt and sign a message.
 [FAIL] The private key cannot be used to decrypt a message
  [HELP] Make sure that the server private key is valid and that there is no passphrase.

 Application configuration

 [PASS] Using latest passbolt version (1.6.5)
 [PASS] Passbolt is configured to force SSL use
 [PASS] App.fullBaseUrl is set to HTTPS
 [PASS] Selenium API endpoints are disabled.
 [PASS] Search engine robots are told not to index content.
 [PASS] Registration is closed, only administrators can add users.
 [PASS] Serving the compiled version of the javascript app
 [PASS] All email notifications will be sent.

 Development Tools (optional)

 [PASS] Phpunit is installed
 [PASS] Phpunit version is 3.7.38

  6 error(s) found. Hang in there!

I hope you can help me I'm really stuck with this one.

**Comment by jaypi95** [comment]
Okay I generated a new key and now I get
[FAIL] The server key fingerprint doesn't match the one defined in app/config.php.


And yes I changed the key in the app.php (there is no config.php).

I also already copied everything fomr /root/.gnupg to /home/www-data/.gnupg

**Comment by ajcomputerses** [comment]
I have the same problem on CentOS 7

¿Any solution? 

Thanks!!!!

**Comment by cedricalfonsi** [comment]
Hello @jaypi95,

> And yes I changed the key in the app.php (there is no config.php)
Can you confirm me that the following variables are correct : 
- Gpg.serverKey.private
- Gpg.serverKey.public
- Gpg.serverKey.fingerprint

> I also already copied everything fomr /root/.gnupg to /home/www-data/.gnupg
It is not required as your key will be added to the keyring by the install script of pasbolt.

> [FAIL] The private key cannot be used to decrypt a message
We've also seen in very rare case the healthcheck rule you mentioned failed, but it wasn't an issue to install passbolt (we are investigating this case). Did you give a try to the install script ?

**Comment by cedricalfonsi** [comment]
>[FAIL] Could not reach the /healthcheck/status with the url specified in App.fullBaseUrl

Regarding this issue, it's relative to your apache/nginx configuration.
What's your flavor ?
nginx same issue : https://github.com/passbolt/passbolt_api/issues/15
apache : enable the mod_rewrite

**Comment by ajcomputerses** [comment]
I can install Passbolt but when i´m triying to register an user i have that problem. 
![2017-10-26 15_46_49-login _ passbolt](https://user-images.githubusercontent.com/18195312/32056419-e89e3bb2-ba64-11e7-9510-f8ef22e40c33.png)



and my healthcheck is this: 

---------------------------------------------------------------
 Healthcheck shell        
---------------------------------------------------------------

 Environment

 [PASS] PHP version 7.0.24
 [PASS] PCRE compiled with unicode support
 [PASS] The temporary directory and its content are writable
 [PASS] The public image directory and its content are writable

 Config files

 [PASS] The core config file is present
 [PASS] The database config file is present
 [PASS] The email config file is present
 [PASS] The application config file is present

 Core config

 [FAIL] Debug mode is on.
  [HELP] Set Configure::write('debug', 0); in app/Config/core.php
 [PASS] Cache is working.
 [PASS] Unique value set for security.salt
 [PASS] Unique value set for security.cipherSeed
 [PASS] Full base url is set to http://10.252.1.156
 [PASS] App.fullBaseUrl validation OK.
 [PASS] /healthcheck/status is reachable.

 SSL Certificate

 [PASS] SSL peer certificate validates
 [PASS] Hostname is matching in SSL certificate.
 [PASS] Not using a self-signed certificate

 Database

 [PASS] Configured to use a supported database backend
 [PASS] The application is able to connect to the database
 [PASS] Not using a prefix for database tables
 [PASS] 20 tables found
 [PASS] Some default content is present
 [PASS] The database schema up to date.

 GPG Configuration

 [PASS] PHP GPG Module is installed and loaded
 [PASS] The server gpg key is not the default one
 [PASS] The environment variable GNUPGHOME is set to /var/cache/nginx/.gnupg
 [PASS] The directory /var/cache/nginx/.gnupg containing the keyring is writable by the user the webserver is running as.
 [PASS] The public key file is defined in app/config.php and readable.
 [PASS] The private key file is defined in app/config.php and readable.
 [PASS] The server key fingerprint matches the one defined in app/config.php.
 [PASS] The server key defined in the app/Config.php is in the keyring.
 [PASS] There is a valid email id defined for the server key.
 [FAIL] The public key cannot be used to encrypt and sign a message
  [HELP] Make sure that the server public key is valid and that there is no passphrase.

 Application configuration

 [PASS] Using latest passbolt version (1.6.5)
 [FAIL] Passbot is not configured to force SSL use
  [HELP] Set App.ssl.force to true in app/Config/app.php
 [FAIL] App.fullBaseUrl is not set to HTTPS
  [HELP] Check App.fullBaseUrl url scheme in app/Config/core.php
 [PASS] Selenium API endpoints are disabled.
 [PASS] Search engine robots are told not to index content.
 [WARN] Registration is open to everyone.
  [HELP] Make sure this instance is not publicly available on the internet.
  [HELP] Or set App.registration.public to false in app/Config/app.php
 [PASS] Serving the compiled version of the javascript app
 [PASS] All email notifications will be sent.

 Development Tools (optional)

 [PASS] Phpunit is installed
 [PASS] Phpunit version is 3.7.38

  4 error(s) found. Hang in there!

**Comment by cedricalfonsi** [comment]
Hello @ajcomputerses,

Can you confirm that the message returned by the healthcheck is not relevant in your case ?
> [HELP] Make sure that the server public key is valid and that there is no passphrase.

It can happen on some environments that the entropy is to low. Gpg requires a certain amount of entropy to work correctly. Can you check the entropy of your system ?
> cat /proc/sys/kernel/random/entropy_avail

People who already have encountered this issue have solved it installing the package rng-tools. 

If it doesn't solve your issue, we will check into the logs.

**Comment by ajcomputerses** [comment]
Thanks for your help. I have installed rng-tools and i have 3127 of entropy_avail. 

I can install and run PassBolt with this problem, but i cannot finalize the user registration. 

Thanks!

**Comment by cedricalfonsi** [comment]
Do you mind to send us some logs/debug data
- app/tmp/logs/error.log of your passbolt;
- your browser console log when trying to login to passbolt, and also when you're completing the plugin setup.
Hhide confidential data if any.

**Comment by ajcomputerses** [comment]
I have revised the error log and there is not information about the problem.

**Comment by jaypi95** [comment]
@cedricalfonsi 

` 'GPG' => [
                // Tell GPG where to find the keyring
                // Needs to be available by the user the webserver is running as
                'env' => [
                        // You can set this to true if you want to customize the location of the keyring.
                        // If false, it will use by default the keyring of the webserver user ~/.gnupg.
                        'setenv' => true,
                        // otherwise you can set the location here
                        // typically on Centos it would be in '/usr/share/httpd/.gnupg'
                        'home' => '/home/www-data/.gnupg',
                ],
                // Main server key
                'serverKey' => [
                        // Server private key location and fingerprint
                        'fingerprint' => '8E74FCCBA454413E7761B937260CD21F63490B79',
                        'public' => APP . 'Config' . DS . 'gpg' . DS . 'public.key',
                        'private' => APP . 'Config' . DS . 'gpg' . DS . 'private.key',
`
This seems correct, doesn't it? 
The public and private key are both in /var/www/passbolt/app/Config/gpg


Alright, I saw this as a possible solution somewhere because I generated the key with the root account instead of www-data

Yes I tried the install script and it installs successfully. However, when I try to access the webpage I get an error 404. It can't find /users/login

The Healthcheck also mentions that it doesn't find any database tables or default content but when I log in to the database myself I'm able to see the tables and content which the install script created.

Regarding the App.fullBaseURL:
I'm not sure what you mean by flavour. I'm using Apache 2.4.18 on Ubuntu 16.04.3 LTS.
mod_rewrite is enabled.

**Comment by cedricalfonsi** [comment]
@ajcomputerses your issue looks different from the issue of @jaypi95 who opened this ticket.
Can you open a new issue and give us more information :
- Browser type and version
- Browser console log (not the tab console log)
- Ensure that you haven't created a server gpg key with a password (passphrase)

**Comment by cedricalfonsi** [comment]
@jaypi95 Looks like you have an issue with your vhost, did you see the documentation we wrote sometimes ago regarding the installation of passbolt on debian 8 w/ apache ?
https://medium.com/passbolt/passbolt-on-debian-8-71-from-scratch-4438dad18908

**Comment by jaypi95** [comment]
Yes I saw that. I just looked at it again and saw that I made a few mistakes with the vhost configuration but it didn't change anything.

**Comment by stripthis** [comment]
@jaypi95 @ajcomputerses @cedricalfonsi This issue is getting real messy and hard to follow. Please reopen an issue on the new community support forum: https://community.passbolt.com/c/installation-issues
